### PR TITLE
Clean up bctest.py and bitcoin-util-test.py

### PR DESCRIPTION
**Pull request by jnewbery**
Apologies for code churn, but I've updated almost all of the code in these files, so I thought I might as well tidy up the code:

- remove newlines
- change tabs for spaces, to align with convention in other py files
- add comments
- add 'Bitcoin Core Developers' copyright notice

### CODE REVIEW

**Comment by MarcoFalke** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/dd76c14977eafc7f09f98f4b40475970c7c74845?w=

**Comment by laanwj** [comment]
> change tabs for spaces, to align with convention in other py files

Oh yes that was irking me, had to change editor settings all the time just for that file. utACK

File: src/test/bitcoin-util-test.py [hunk 1]
```diff
@@ -16,11 +17,13 @@
 
 Can also be run manually from the src directory by specifiying the source directory:
 
-test/bitcoin-util-test.py --src=[srcdir]
+test/bitcoin-util-test.py --src='srcdir' [--verbose]
```
Comment (laanwj) [inline src/test/bitcoin-util-test.py]: I think this comment is wrong, too? It looks for `--srcdir` not `--src`.

File: src/test/bctest.py [hunk 2]
```diff
@@ -11,94 +12,108 @@
 import logging
 
 def parse_output(a, fmt):
-	if fmt == 'json': # json: compare parsed data
-		return json.loads(a)
-	elif fmt == 'hex': # hex: parse and compare binary data
-		return binascii.a2b_hex(a.strip())
-	else:
-		raise NotImplementedError("Don't know how to compare %s" % fmt)
+    """Parse the output a with format fmt. fmt can either be hex or json.
```
Comment (MarcoFalke) [inline src/test/bctest.py]: µNit: no need to mention what fmt can be, when it is mentioned in the code just a line below.

**Comment by jnewbery** [comment]
Nits addressed and commits squashed.

File: src/test/bitcoin-util-test.py (hunk 1 above)
Comment (jnewbery) [inline src/test/bitcoin-util-test.py]: Good spot. Thanks.

File: src/test/bctest.py (hunk 2 above)
Comment (jnewbery) [inline src/test/bctest.py]: µACK. Comment shortened.

**Comment by fanquake** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/9069/commits/2b175d4b018be432e333bf8a343b85c5f284264f
The non-whitespace diff is seen with [?w=1](https://github.com/bitcoin/bitcoin/pull/9069/commits/2b175d4b018be432e333bf8a343b85c5f284264f?w=1).

**Pull request merged**